### DISCUSSION TITLE
Unable to set Vue Router into History mode

### DISCUSSION THREAD

**Discussion by jmriyaz84** [quasarframework/quasar#9769] [General - Components / Directives / etc] [answered]
Hi,
I want to change vue router's hash mode to history to avoid # in the url , I have changed the mode it is working fine with quasar dev , but when I deploy into Apache Server I am getting 404 , 
I have followed below link [https://router.vuejs.org/guide/essentials/history-mode.html#example-server-configurations](url) here , I not able to add following code in httpd.config file , I can't use .htaccess due to performance & security issues.
_<IfModule mod_rewrite.c>
  RewriteEngine On
  RewriteBase /
  RewriteRule ^index\.html$ - [L]
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteRule . /index.html [L]
</IfModule>_
Please guide how to add redirect rule in Apache's httpd.conf file , I am deploying my code under /var/www/html folder.

**Comment by ontwikkelfabriek** [comment, marked as answer]
@jmriyaz84 

> RewriteBase /

Did you install & enable the `mod_rewrite` module in Apache?

You can't just copy the sample config and expect to work. You have to adjust to match the path where your Quasar app is located:
https://stackoverflow.com/questions/30193087/mod-rewrite-without-htaccess

I suggest you try Nginx instead of Apache(or even Caddy2). Much simpler setup and way more performant.

btw Quasar in dev mode is using a Nodejs(express?) server.
